### PR TITLE
Refactored node validations into classes 

### PR DESCRIPTION
**Pull request by micapam**
As per SITES-375.
Also added a validator to prevent recursive ancestry (SITES-355).

### CODE REVIEW

File: app/models/news_article.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
 class NewsArticle < Node
   store_attribute :data, :release_date, Date
-end
\ No newline at end of file
+
+  validates_presence_of :section, :parent
```
Comment (ltankey) [inline app/models/news_article.rb]: News Articles can be created without a section, and likely won't have parents.

**Comment by maxious** [comment]
LGTM

**Comment by ltankey** [comment]
Aside from the validation on News Articles (which we would likely need to change in a future PR), LGMT

File: app/models/news_article.rb (hunk 1 above)
Comment (micapam) [inline app/models/news_article.rb]: Yeah I kind of dithered over whether to include this - and decided that I'd leave it up to the developer implementing NewsArticle to remove these validation(s) if they see fit.

**Comment by micapam** [comment]
bump for status?

**Comment by ltankey** [comment]
In that case LGTM

**Pull request merged**